common/management/commands/volunteer_renewal_reminders.py

```diff
@@ -0,0 +1,94 @@
+
+        now = timezone.now()
+
+        from civictechprojects.models import VolunteerRelation
+        volunteer_applications = VolunteerRelation.objects.all()
+        for volunteer_relation in volunteer_applications:
+            if volunteer_relation.is_up_for_renewal(now):
+                if now > volunteer_relation.projected_end_date:
+                    if volunteer_relation.is_approved:
+                        # Don't send conclusion emails if volunteer wasn't approved
+                        send_volunteer_conclude_email(volunteer_relation.volunteer, volunteer_relation.project.project_name)
+                        notify_project_owners_volunteer_concluded_email(volunteer_relation)
+                    volunteer_relation.delete()
+                elif volunteer_relation.is_approved:
+                    # Don't send reminders if volunteer isn't approved
+                    email_template = get_reminder_template_if_time(now, volunteer_relation)
+                    if email_template:
+                        send_reminder_email(email_template, volunteer_relation)
+                        volunteer_relation.re_enroll_reminder_count += 1
+                        volunteer_relation.re_enroll_last_reminder_date = now
+                        volunteer_relation.save()
+
+
+def get_reminder_template_if_time(now, volunteer):
+    reminder_interval_days = settings.VOLUNTEER_RENEW_REMINDER_PERIODS
+
+    if not volunteer.re_enroll_last_reminder_date:
+        return volunteer_reminder_emails[0]
+    else:
+        days_since_last_reminder = (now - volunteer.re_enroll_last_reminder_date).days
+        days_to_next_reminder = reminder_interval_days[min(volunteer.re_enroll_reminder_count, len(reminder_interval_days) - 1)]
+        return (days_to_next_reminder > 0) and (days_since_last_reminder >= days_to_next_reminder) and volunteer_reminder_emails[volunteer.re_enroll_reminder_count]
+
+
+def send_reminder_email(email_template, volunteer_relation):
+    project = volunteer_relation.project
+    volunteer = volunteer_relation.volunteer
+    context = {
+        'first_name': volunteer.first_name,
+        'project_name': project.project_name,
+        'project_end_date': datetime_to_string(datetime_field_to_datetime(volunteer_relation.projected_end_date), DateTimeFormats.DATE_LOCALIZED),
+        'volunteer_start_date': datetime_to_string(datetime_field_to_datetime(volunteer_relation.application_date), DateTimeFormats.DATE_LOCALIZED)
+    }
+
+    email_msg = EmailMessage(
+        subject="You're making a difference at " + project.project_name,
+        from_email=_get_account_from_email(settings.EMAIL_VOLUNTEER_ACCT),
+        to=[volunteer.email],
+    )
+
+    email_msg = email_template.render(email_msg, context)
+    send_email(email_msg, settings.EMAIL_VOLUNTEER_ACCT)
+
+
+review_commitment_url = settings.PROTOCOL_DOMAIN + '/index/?section=MyProjects&from=renewal_notification_email'
+
+
+def get_first_email_template():
```

Comment:
I mentioned this in standup but I really like this method of making the python readable when HTML email frequently is not. 👍